libs/core/src/main/java/org/opensearch/core/common/io/stream/BaseWriteable.java

```diff
@@ -28,6 +28,7 @@
      */
     class WriteableRegistry {
         private static final Map<Class<?>, Writer<? extends BaseStreamOutput, ?>> WRITER_REGISTRY = new ConcurrentHashMap<>();
+        private static final Map<Class<?>, Class<?>> WRITER_CUSTOM_CLASS_MAP = new ConcurrentHashMap<>();
```

Comment:
I struggle to understand what purpose it serves, there is only one example in the code

```
WriteableRegistry.registerWriterCustomClass(ReadableInstant.class, ReadableInstant.class);
```

Is it like "if this class X is used in the code the threat is as class Y`?

Reply:
Correct. Per t[his javadoc](https://github.com/opensearch-project/OpenSearch/pull/7780/files#diff-bc10bafd3fdd4fb0529c639b7276b0f238b60be22654d2c05e5540caf5e3a314R85)

If the value to write is an instanceof the registered class type then return the writer registered to that type. This to cover class scope without registering all child classes explicitly.

Reply:
And there's a typo there I'll fix /s/instant/instance